lib/nghttp2_hd.c

```diff
@@ -1654,8 +1674,9 @@ static ssize_t hd_inflate_read_len(nghttp2_hd_inflater *inflater, int *rfin,
   }
 
   if (out > maxlen) {
-    DEBUGF("inflatehd: integer exceeded the maximum value %zu\n", maxlen);
-    return NGHTTP2_ERR_HEADER_COMP;
+    DEBUGF("inflatehd: integer %zu exceeded the maximum value %zu\n", out, maxlen);
+    inflater->left = out;
+    return NGHTTP2_ERR_HEADER_COMP_MAX_LEN_EX;
```

Comment:
Maybe the tests failed because they expected `inflater->left` to be 0 (or whatever initial state it was) and a return value of `NGTTP2_ERR_HEADER_COMP`?